src/hotspot/share/gc/shenandoah/shenandoahMmuTracker.hpp

```diff
@@ -68,22 +84,22 @@ class ShenandoahMmuTracker {
   // This enrolls the periodic task after everything is initialized.
   void initialize();
 
-  // This is called at the start and end of a GC cycle. The GC thread times
-  // will be accumulated in this generation. Note that the bootstrap cycle
-  // for an old collection should be counted against the old generation.
-  // When the collector is idle, it still runs a regulator and a control.
-  // The times for these threads are attributed to the global generation.
-  void record(ShenandoahGeneration* generation);
+  // At completion of each GC cycle (not including interrupted cycles), we invoke one of the following to record the
+  // GC utilization during this cycle.
+  //
+  // We may redundantly record degen and full, in which case the gcid will repeat.  We log these as FULL.
```

Comment:
Won't `gc_id` also repeat for old_marking_increments too?

Reply:
Or may be I am misunderstanding what "repeat" means here.

Reply:
I'm updating this comment on the expand-old-on-demand branch to try to make this more clear.